odp_sdk/tabular_storage_client.py

```diff
@@ -0,0 +1,268 @@
+from typing import Dict, List, Optional
+from uuid import UUID
+
+import requests
+from pydantic import BaseModel, PrivateAttr
+
+from odp_sdk.dto import ResourceDto
+from odp_sdk.dto.table_spec import StageDataPoints, TableSpec
+from odp_sdk.dto.tabular_store import TableStage
+from odp_sdk.exc import OdpResourceExistsError, OdpResourceNotFoundError
+from odp_sdk.http_client import OdpHttpClient
+from odp_sdk.oqs.oqs_base import OqsBasePredicate
+
+
+class OdpTabularStorageController(BaseModel):
+    _http_client: OdpHttpClient = PrivateAttr(default_factory=OdpHttpClient)
+    tabular_storage_endpoint: str = "/data"
+
+    @property
+    def tabular_storage_url(self) -> str:
+        """
+        The URL of the tabular storage endpoint, including the base URL.
+
+        Returns:
+            The tabular storage URL
+        """
+        return f"{self._http_client.base_url}{self.tabular_storage_endpoint}"
+
+    def create_schema(self, resource_dto: ResourceDto, table_spec: TableSpec) -> TableSpec:
```

Comment:
In the raw controller i might have gone a bit overboard with how the users can reference datasets. Like [this example](https://github.com/C4IROcean/odp-sdk-python/blob/6cdbc5bcd71ea94e5f9931c8f13aba2fcf62f224/odp_sdk/raw_storage_client.py#L38). But I liked the idea of making it simpler by allowing the user to use name of dataset, uuid of dataset or the resourcedto. Maybe we should agree on one way of doing it so we have some consistency for the users?

Reply:
I think since API already has the functionality, we don't need to restrict the SDK. Why do you think we should make it simpler?

Reply:
I think that using a ResourceDto as a reference to a resource could be a nice addition to referencing by UUID and Qualified name. I think that makes it more flexible that restrictive?

Reply:
I think unification in our case is more maintainable, just a simple object creation is not that cumbersome but maintaining all the other possibilities as I see it is not worth the QoL improvement. In the future if we decide to add the functionality it wouldn't be a problem.